modules/widgets/social-icons.php

```diff
@@ -0,0 +1,500 @@
+					<option value="<?php echo esc_attr( $value ); ?>" <?php selected( $value, $instance['icon-size'] ); ?>><?php echo esc_attr( $label ); ?></option>
+				<?php endforeach; ?>
+			</select>
+		</p>
+
+		<div class="jetpack-social-icons-widget-list"
+			data-url-icon-id="<?php echo $this->get_field_id( 'url-icons' ); ?>"
+			data-url-icon-name="<?php echo $this->get_field_name( 'url-icons' ); ?>"
+		>
+
+			<?php
+				foreach ( $instance['icons'] as $icon ) {
+					self::render_icons_template( array(
+						'url-icon-id'   => $this->get_field_id( 'url-icons' ),
+						'url-icon-name' => $this->get_field_name( 'url-icons' ),
+						'url-value'     => $icon['url'],
+					) );
+				}
+			?>
+
+		</div>
+
+		<p class="jetpack-social-icons-widget add-button">
+			<button type="button" class="button jetpack-social-icons-add-button">
+				<?php esc_html_e( 'Add an icon', 'jetpack' ); ?>
+			</button>
+		</p>
+
+		<?php
+			$support     = 'https://en.support.wordpress.com/widgets/social-media-icons-widget/#available-icons';
+			if ( 'es' === get_locale() ) {
+				$support = 'https://es.support.wordpress.com/social-media-icons-widget/#iconos-disponibles';
+			}
+			if ( 'pt-br' === get_locale() ) {
+				$support = 'https://br.support.wordpress.com/widgets/widget-de-icones-sociais/#ícones-disponíveis';
+			}
+		?>
+
+		<p>
+			<em><a href="<?php echo esc_url( $support ); ?>" target="_blank">
+				<?php esc_html_e( 'View available icons', 'jetpack' ); ?>
+			</a></em>
+		</p>
+
+	<?php
+	}
+
+	/**
+	 * Generates template to add icons.
+	 *
+	 * @param array $args Template arguments
+	 */
+	static function render_icons_template( $args = array() ) {
+		$defaults = array(
+			'url-icon-id'   => '',
+			'url-icon-name' => '',
+			'url-value'     => '',
+		);
+
+		$args = wp_parse_args( $args, $defaults );
+		?>
+
+		<div class="jetpack-social-icons-widget-item">
+			<div class="jetpack-social-icons-widget-item-wrapper">
+				<div class="handle"></div>
+
+				<p class="jetpack-widget-social-icons-url">
+					<?php
+						printf( '<input class="widefat id="%1$s" name="%2$s[]" type="text" placeholder="%3$s" value="%4$s"/>',
+							esc_attr( $args['url-icon-id'] ),
+							esc_attr( $args['url-icon-name'] ),
+							esc_attr__( 'Account URL', 'jetpack' ),
+							esc_url( $args['url-value'] )
+						);
+					?>
+				</p>
+
+				<p class="jetpack-widget-social-icons-remove-item">
+					<a class="jetpack-widget-social-icons-remove-item-button" href="javascript:;">
+						<?php esc_html_e( 'Remove', 'jetpack' ); ?>
+					</a>
+				</p>
+			</div>
+		</div>
+
+		<?php
+	}
+} // Jetpack_Widget_Social_Icons
+
+/**
+ * Register and load the widget.
+ *
+ * @access public
+ * @return void
+ */
+function jetpack_widget_social_icons_load() {
+	register_widget( 'Jetpack_Widget_Social_Icons' );
+}
+add_action( 'widgets_init', 'jetpack_widget_social_icons_load' );
+
+if ( ! function_exists( 'jetpack_social_menu_get_svg' ) ) :
```

Comment:
Why is this in a function_exists?  Do we have an alternate version wpcom side or the like?